src/http_client.ts

```diff
@@ -131,6 +133,15 @@ export class HttpClient {
         });
     }
 
+    private _informAboutStreamNoRetry() {
+        // using a module-global variable to only warn once per whole process
+        if (!warnedAboutStreamNoRetry) {
+            warnedAboutStreamNoRetry = true;
+            this.logger.warning('Request body was a stream - retrying will not work, as part of it was already consumed.');
```

Comment:
The `logger.warning` function will automatically send the same message just once so you can utilize that :)

Reply:
:open_mouth: TIL, thanks!